src/mongocxx/model/write.cpp

```diff
@@ -21,23 +21,56 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace model {
 
 write::write(insert_one value) : _type(write_type::k_insert_one), _insert_one(std::move(value)) {
+    static_assert(std::is_nothrow_move_constructible<insert_one>::value, "Move-construct may throw");
```

Comment:
You should include `<type_traits>` above so that the is_xxx things are definitely defined.